test/datatable.jl

```diff
@@ -341,4 +341,26 @@ module TestDataTable
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), :A).columns) == [1]
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), [1]).columns) == [1]
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), 1).columns) == [1]
+
+    @testset "duplicate entries in unstack warnings" begin
+        dt = DataTable(id=NullableArray([1, 2, 1, 2]), variable=["a", "b", "a", "b"], value=[3, 4, 5, 6])
+        if isdefined(Symbol(:@test_warn))
```

Comment:
I couldn't get `isdefined` to work because Julia will still try and expand the `@test_warn` macro during parsing. I think it needs a `@static` check to avoid the if block entirely during parsing in v0.5. A static check against the first tagged version with the `@test_warn` macro works, although I'm not sure how to reference the exact commit. https://github.com/JuliaLang/julia/commit/28a11ffd4d2e8451492013f18dad5e9a576233f6